### ISSUE TITLE
Can't Update Report

### ISSUE THREAD

**Issue by smcgregor** [responsible-ai-collaborative/aiid#1280]
I am not sure what is blocking me from updating the report found here: https://incidentdatabase.ai/cite/edit?report_number=1765

The button is not active and there is no apparent message indicating why. I successfully updated another report a moment ago, so there is something about this specific report. I think it needs some user-apparent error message so I will know what is necessary to change before it will allow me to submit the form.

**Comment by smcgregor** [comment]
This is also a problem on staging.

**Comment by LKchemposer** [comment]
It's a problem with many reports

![image](https://user-images.githubusercontent.com/39468345/196372136-a64f633d-d128-462c-8596-74f4df8b60c0.png)

**Comment by pdcp1** [comment]
@smcgregor @LKchemposer This is ready to review (not merged into Staging yet). If you want to take a look and test it from the functional perspective, you can do it in the preview link https://deploy-preview-1292--staging-aiid.netlify.app

I found similar issues on 3 different forms:
- `/cite/edit` report form
- `/apps/submitted` edit submission form
- `/apps/submit` new submission form

I added more information messages about missing/invalid fields to let users know why the `Update` button is disabled.

**Comment by smcgregor** [comment]
The button is enabled now for me, it looks like you also fixed some submit form state issues?

**Comment by pdcp1** [comment]
Yes, I also fixed other form state issues.

Basically, I tried to improve the UX by keeping the user informed about the status of the form state all the time. For instance, in the `/apps/submitted` edit submission form, when the Incident Id is empty, the `Alleged developer`, `Alleged deployers` and `Alleged harmed` fields are required but collapsed in the `Tell us more...` section. So, I added a warning message on the section header to let the user know about those required fields.
Now it's easier to know what is missing just by scrolling through the form.

<img width="866" alt="image" src="https://user-images.githubusercontent.com/6564809/196823315-acee3a40-f50b-4de0-93b8-bb0b58519a15.png">
